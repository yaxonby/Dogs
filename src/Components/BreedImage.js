import React from "react";
import {Component} from "react";
import MyImage from "./function/MyImage"

export default class BreedImage extends Component{
	constructor(props) {
	super(props);
	}

	shouldComponentUpdate(nextProps, nextState, nextContext) {
		console.log("nextProps=", nextProps,"nextState=", nextState, "nextContext=", nextContext)
		return true
	}

	render () {
		const {ListImageBreed, breedName}=this.props
		if (!ListImageBreed) {
		return (<p> </p>)
				} else {
					return (
						<div>
						<h3> View images breed {breedName}  </h3>
							<div>
								{ListImageBreed.map(function(elem, index) {return(
									<li className="liStyle" key={index}>
										<MyImage src={elem}/>
									</li>
								)}
								)}
							</div>
						</div>
					);
					};
	}
}

//<img src={elem}/>
