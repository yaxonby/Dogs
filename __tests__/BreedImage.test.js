import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import BreedImage from '../src/Components/BreedImage';

//Enzyme.configure({ adapter: new Adapter() });



it("check prop is null by default", () => {
    const BreedImageComponent = shallow(<BreedImage />);
    expect(BreedImageComponent.props().dogBreed).toBe(undefined);
});

it('should render a ViewBreedImage', () => {
    const wrapper = shallow(
        <BreedImage />
    );
    expect(wrapper).toMatchSnapshot();
});


describe('News container', () => {
  const props = { // описываем props
dogBreed:[]
    }
    // функция получения новостей onGetNews. Ее содержимое тестировать не нужно.
    // Но нам потребуется протестировать, что функция была вызвана в componentDidMount
  //  onGetNews: () => {},
//  }
  // здесь будут будущие it
})

const props = { dogBreed:[] }

describe('News container initial', () => { // группируем еще на один уровень, так как здесь потом будет тест componentDidMount
	 const newsContainer = shallow(<BreedImage  {...props} />)

	 it('render initial', () => {
		 expect(newsContainer.find('p')).toHaveLength(1) // .find + поиск по тэгу
		// expect(newsContainer.find('div')).toHaveLength(0) // .find + поиск по имени компонента
		// expect(newsContainer.find('p').text()).toEqual('Loading...')

	 })

 })
