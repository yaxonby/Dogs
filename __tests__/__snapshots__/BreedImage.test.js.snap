// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a ViewBreedImage 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BreedImage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": " ",
    },
    "ref": null,
    "rendered": " ",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": " ",
      },
      "ref": null,
      "rendered": " ",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
